CMakeLists.txt

```diff
@@ -54,17 +54,36 @@
 
 cmake_minimum_required(VERSION 3.1.0)
 
-set(CMAKE_MODULE_PATH Builds/CMake)
+if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
+  set(dir "build")
+  set(cmd "cmake")
+  if (target)
+    set(dir "${dir}/${target}")
+    set(cmd "${cmd} -Dtarget=${target}")
+  elseif(CMAKE_BUILD_TYPE)
+    set(dir "${dir}/${CMAKE_BUILD_TYPE}")
+    set(cmd "${cmd} -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}")
+  else()
+    set(dir "${dir}/default")
+  endif()
+  set(cmd "${cmd} ../..")
+
+  message(FATAL_ERROR "In-source builds are not allowed. Instead try:\n"
+    "rm CMakeCache.txt && mkdir -p ${dir} && cd ${dir} && ${cmd}")
```

Comment:
We're suggesting commands, not running them, so I'd remove `CMakeFiles` too. The goal is "remove the cruft that cmake created".

We could also just tell them "remove the CMakeCache.txt file and CMakeFiles directory" without giving them a command.